### DISCUSSION TITLE
Default template configuration

### DISCUSSION THREAD

**Discussion by fungs** [filips123/PWAsForFirefox#413] [Q&A]
Hi,

I've seen some configuration injection features on the release notes <- 2.8.0 but I'm unsure if that would solve my usability issues. When creating from-scratch new web app environments using the standard procedure, there are suboptimal pre-configurations. I'll list some of them but there should be many more, depending on host system and user preferences:

1. the default browser content language is set to English and the browser asks you to do offline translation or the web app doesn't show in native language
2. the browser language itself is English, to change that, I have to install a language pack in each environment
3. there are some very basic add-ons, like spell checking, which I also need to install in every environment

Could there be some way to solve these usability issues (including using a default profile to configure and then copy or overlay-mount?

Thanks for the great add-on!

**Comment by filips123** [comment]
For most things, it should work if you create a profile, configure it how you want (either by editing config files or from Firefox), and then copy the profile to another directory. You may also need to delete files such as `parent.lock` and other files that contain unnecessary data (like cookies, history, sessions, etc.), or even copy just the files you want. Which files exactly should that be depends on what you're trying to configure.

Then, you can set that directory as a [default template directory](https://pwasforfirefox.filips.si/user-guide/extension/#default-profile-template) in the extension settings. Then, when you create a new profile, all files in the template directory will be copied to a newely created profile. 

Except the "default" profile, which currently does not support profile templates (#376). To apply your settings there, you'll need to manually copy your config files.

**Reply by fungs** [reply]
Thanks @filips123 , I didn't know that default template directory feature. That should solve the issues, for me at least. I promise to test and report once I find the time.

For other users, the way to do this might be too technical. Having a way to specify configuration key value pairs for the default profile or specifying add-ons in the UI might be a more intuitive way to go. I know that takes time and effort...

**Reply by filips123** [reply]
Well, the alternative to manually editing files is to configure settings and addons through normal settings UI, and copy files where settings are stored to the template. I don't know how this could be made easier in  good way...

**Reply by fungs** [reply]
A simple idea: a button to open `about:config`, `about:addons` or `about:preferences` in a browser instance with the default template profile loaded (or creating a new one on demand).
